### PR TITLE
fixing how joint names are filled up, fixed joints were getting included...

### PR DESCRIPTION
**Pull request by sachinchitta**
... earlier, also resizing consistency limits for when randompositionsnearby function is being called to account for mimic joints.

### CODE REVIEW

File: planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp [hunk 1]
```diff
@@ -86,7 +86,17 @@ void KDLKinematicsPlugin::getRandomConfiguration(const KDL::JntArray &seed_state
   std::vector<double> near(dimension_, 0.0);
   for (std::size_t i = 0 ; i < dimension_; ++i)
     near[i] = seed_state(i);
-  joint_model_group_->getVariableRandomPositionsNearBy(state_->getRandomNumberGenerator(), values, near, consistency_limits);
+
+  // Need to resize the consistency limits to remove mimic joints
+  std::vector<double> consistency_limits_mimic;
```
Comment (isucan) [inline planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp:91]: This vector should be constructed outside this function and reused.

File: planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp [hunk 2]
```diff
@@ -164,12 +174,14 @@ bool KDLKinematicsPlugin::initialize(const std::string &robot_description,
...
-  for (std::size_t i = 0; i < joint_model_group->getJointModels().size(); ++i)
+  for (std::size_t i=0; i < joint_model_group->getJointModels().size(); ++i)
   {
-    const std::vector<moveit_msgs::JointLimits> &jvec = joint_model_group->getJointModels()[i]->getVariableBoundsMsg();
-    ik_chain_info_.limits.insert(ik_chain_info_.limits.end(), jvec.begin(), jvec.end());
+    if(joint_model_group->getJointModels()[i]->getType() == moveit::core::JointModel::REVOLUTE || joint_model_group->getJointModels()[i]->getType() == moveit::core::JointModel::PRISMATIC)
+    {
+      ik_chain_info_.joint_names.push_back(joint_model_group->getJointModelNames()[i]);
+      const std::vector<moveit_msgs::JointLimits> &jvec = joint_model_group->getJointModels()[i]->getVariableBoundsMsg();
+      ik_chain_info_.limits.insert(ik_chain_info_.limits.end(), jvec.begin(), jvec.end());
```
Comment (isucan) [inline planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp:183]: You can also do:

```
 ik_chain_info_.limits.push_back(joint_model_group->getJointModels()[i]->getVariableBoundsMsg()[0]);
```

File: planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp (hunk 1 above)
Comment (sachinchitta) [inline planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp:91]: As a member variable? This function and the functions that call it are all supposed to be const.

Comment (isucan) [inline planning/kdl_kinematics_plugin/src/kdl_kinematics_plugin.cpp:91]: Nevermind, I misread this. I missed that consistency_limits was an argument to the function, not a member variable. Sorry about that!

**Pull request merged**